internal/daemon/worker/worker.go

```diff
@@ -279,6 +284,33 @@ func (w *Worker) Start() error {
 			return fmt.Errorf("error loading worker auth creds: %w", err)
 		}
 
+		// Don't output a fetch request if an activation token has been
+		// provided. Technically we _could_ still output a fetch request, but if
+		// a token was provided it may well be confusing if it seems like it was
+		// ignored because a fetch request was still output.
+		//
+		// Additionally, if we have an activation token and the nonce doesn't
+		// match, update the stored nonce. This could happen if, for instance, a
+		// worker was created in the cluster via the API, then removed, then
+		// submitted to the worker. It won't be valid any longer, so this allows
+		// the operator to update the value.
+		if actToken := w.conf.RawConfig.Worker.ControllerGeneratedActivationToken; actToken != "" {
+			createFetchRequest = false
```

Comment:
What if a user starts a worker with an activation token, it comes up and auths, then it's later taken down for maintenance and then brought back up? Are we expecting them to remove the original activation token from the config?

Reply:
No -- this block will ensure that the stored nonce matches what's in the config file, if the creds are already in existence. However, in `protocol.Dial`, called from `controllerDialerFunc` in `controller_connection.go`, we only attempt a fetch if we do not already have certificates. After the initial fetch we will have certificates, so we won't actually make use of this value.

Reply:
(Note that that's the same behavior for worker-led, as well, which is why it stops trying to fetch after it successfully does.)

Reply:
I'm going to make a change here though because nodee by default will clear the nonce so this results in an extra write. Instead I'll pass it through to the dial function in an option and it can be used there if needed.

Reply:
I've made this change in https://github.com/hashicorp/boundary/pull/2413/commits/a0eb50e4835fb8ceaa357aefa7bcd4c8f8ff3d16